packages/core/src/awsService/appBuilder/wizards/serverlessLandWizard.ts

```diff
@@ -0,0 +1,290 @@
+}
+
+export interface ProjectMetadata {
+    patterns: Record<string, PatternData>
+}
+
+/**
+ * Manages metadata for serverless application patterns
+ * Implements Singleton pattern to ensure single instance across the application
+ */
+export class MetadataManager {
+    private static instance: MetadataManager
+    private metadata: ProjectMetadata | undefined
+
+    private constructor() {}
+
+    public static getInstance(): MetadataManager {
+        if (!MetadataManager.instance) {
+            MetadataManager.instance = new MetadataManager()
+        }
+        return MetadataManager.instance
+    }
+
+    /**
+     * Loads metadata from a JSON file
+     * @param metadataPath Path to the metadata JSON file
+     * @returns Promise containing the parsed ProjectMetadata
+     */
+    public async loadMetadata(metadataPath: string): Promise<ProjectMetadata> {
+        try {
+            if (!this.metadata) {
+                const metadataContent = nodefs.readFileSync(metadataPath, { encoding: 'utf-8' })
+                const parseMetadata = JSON.parse(metadataContent) as ProjectMetadata
+                this.metadata = parseMetadata
+            }
+            return this.metadata
+        } catch (err) {
+            throw new Error(`Failed to load metadata: ${err instanceof Error ? err.message : String(err)}`)
+        }
+    }
+
+    /**
+     * Retrieves available patterns with their descriptions
+     * @returns Array of pattern objects containing labels and descriptions
+     */
+    public getPatterns(): { label: string; description?: string }[] {
+        if (!this.metadata) {
+            return []
+        }
+        return Object.entries(this.metadata.patterns).map(([patternName, patternData]) => {
+            let description: string | undefined = undefined
+            if (typeof patternData === 'string') {
+                description = patternData
+            } else if (Array.isArray(patternData)) {
+                // If description is an array, join it into a single string
+                description = patternData.join(' ')
+            }
+            if (!patternData || !patternData.name) {
+                return {
+                    label: patternName,
+                    description: description || 'No description available',
+                }
+            }
+            return {
+                label: patternName,
+                description: patternData.description,
+            }
+        })
+    }
+
+    /**
+     * Gets available runtimes for a specific pattern
+     * @param pattern The pattern name to get runtimes for
+     * @returns Array of runtime options with labels
+     */
+    public getRuntimes(pattern: string): { label: string }[] {
+        const patternData = this.metadata?.patterns?.[pattern]
+        if (!patternData || !patternData.runtimes) {
+            return []
+        }
+        return patternData.runtimes.map((runtime) => ({
+            label: runtime.name,
+        }))
+    }
```

Comment:
This would need to change from what we discussed. The list of runtimes would probably not be just as a list, but you need to take it (and group it) from the list of available options (multiple combinations runtime+IaC)

Reply:
I have those changes ready, will publish a new PR for that